UnitTest.Rollbar/RollbarLoggerFixture.cs

```diff
@@ -38,8 +77,81 @@ public void ReportFromCatch()
             }
             catch (System.Exception ex)
             {
-                _logger.Error(new System.Exception("outer exception", ex));
+                try
+                {
+                    _logger.Error(new System.Exception("outer exception", ex));
+                }
+                catch
+                {
+                    Assert.IsTrue(false);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void ReportMessage()
+        {
+            try
+            {
+                _logger.Log(ErrorLevel.Error, "test message");
             }
+            catch
+            {
+                Assert.IsTrue(false);
+            }
+        }
+
+
+        private const int maxCallLengthInMillisec = 500;
+
+        [TestMethod]
+        [Timeout(maxCallLengthInMillisec)]
+        public void LongReportIsAsync()
+        {
+            RollbarConfig loggerConfig =
+                new RollbarConfig(RollbarUnitTestSettings.AccessToken) { Environment = RollbarUnitTestSettings.Environment, };
+            loggerConfig.Transform = delegate { System.Threading.Thread.Sleep(TimeSpan.FromMilliseconds(10 * maxCallLengthInMillisec)); };
+            IRollbar logger = RollbarFactory.CreateNew().Configure(loggerConfig);
+
+            try
+            {
+                logger.Log(ErrorLevel.Error, "test message");
+            }
+            catch
+            {
+                Assert.IsTrue(false);
+            }
+        }
+
+        [TestMethod]
+        public void ExceptionWhileTransformingPayload()
+        {
+            this._transformException = false;
+
+            RollbarConfig loggerConfig =
+                new RollbarConfig(RollbarUnitTestSettings.AccessToken) { Environment = RollbarUnitTestSettings.Environment, };
+            loggerConfig.Transform = delegate { throw new NullReferenceException(); };
+            IRollbar logger = RollbarFactory.CreateNew().Configure(loggerConfig);
+            logger.InternalEvent += Logger_InternalEvent;
+
+            try
+            {
+                logger.Log(ErrorLevel.Error, "test message");
+            }
+            catch
+            {
+                Assert.IsTrue(false);
+            }
+
+            System.Threading.Thread.Sleep(TimeSpan.FromMilliseconds(3 * maxCallLengthInMillisec));
```

Comment:
These tests that depend on timeouts are fragile, can we do something else using a callback or semaphore?

Reply:
good point! corrected.